### PR TITLE
Fix various sanitycheck problems on nrf52840_pca10056

### PR DESCRIPTION
**Pull request by andyross**
This board doesn't pass the full test suite right now.  A few of the problems were fairly simple to address.

### CODE REVIEW

File: samples/subsys/power/device_pm/sample.yaml [hunk 1]
```diff
@@ -4,3 +4,15 @@ tests:
   sample.power.ospm.dev_idle_pm:
     platform_whitelist: nrf52840_pca10056 nrf52_pca10040
     tags: power
+    harness: console
+    harness_config:
+      type: multi_line
+      regex:
+        - "Device PM sample app start"
+        - "parent resuming\\.\\."
```
Comment (marc-hb) [inline samples/subsys/power/device_pm/sample.yaml]: The double backslashes suck, these should really be Python "raw" strings. Could they be? Not in this PR I guess but at some point in the future.

**Review by marc-hb — APPROVED** [review]
Note there's a very generic commit and important sanitycheck bug fix "hidden" in this apparently platform specific PR.

+1 assuming all the corresponding tests pass.

Is there some nightly build where these tests were regularly failing and will start passing? Or are they only ever run manually and condemned to bitrot again in the future?

**Review by pizi-nordic — APPROVED** [review]

File: samples/subsys/usb/dfu/sample.yaml [hunk 2]
```diff
@@ -2,6 +2,7 @@ sample:
   name: USB DFU sample
 tests:
   sample.subsys.usb.dfu:
+    build_only: True
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/dfu/sample.yaml]: nit-pick: s/True/true, it is `build_only: true` everywhere in samples and tests.

**Review by nashif — APPROVED** [review]

File: samples/subsys/power/device_pm/sample.yaml (hunk 1 above)
Comment (andyross) [reply samples/subsys/power/device_pm/sample.yaml]: It's YAML, actually.  The first backslash is required (we're matching a literal period character and not any character), the second is to get it through YAML which also interprets backslash escapes and needs a second one to pass it through to the next layer.

Surely YAML has a better string delimeter that does what we want too, but I don't know it and quite frankly would rather spend my day reading C code in ANSI trigraphs than try to look up YAML syntax. :)

Comment (marc-hb) [reply samples/subsys/power/device_pm/sample.yaml]: I just did lookup YAML syntax briefly. Single quotes should Just Work:
```
regex:
        - 'Device PM sample app start'
        - 'parent resuming\.\.'
```
According to this page http://blogs.perl.org/users/tinita/2018/03/strings-in-yaml---to-quote-or-not-to-quote.html
> A double quoted scalar is the only scalar style where you can use escape (backslash) sequences.

So using double-quotes here seems like shooting yourself in the foot.

**Pull request merged**